src/dream.ml

```diff
@@ -260,9 +260,9 @@ let drop_session_field = Session.drop_session_field
 let all_session_values = Session.all_session_values
 let all_session_fields = all_session_values
 let invalidate_session = Session.invalidate_session
-let memory_sessions = Session.memory_sessions
-let cookie_sessions = Session.cookie_sessions
-let sql_sessions = Sql_session.sql_sessions
+let memory_sessions = Session.memory_sessions ()
+let cookie_sessions = Session.cookie_sessions ()
+let sql_sessions = Sql_session.sql_sessions ()
```

Comment:
At first glance, this looks like whatever is delayed by the extra `unit` parameter, will be triggered once per process, rather than once per server (or once per `scope`). With this change, can there be two concurrent uses of two different `memory_session` middlewares in one process?

Reply:
Thanks for your questions. Right after you answered them, I realised that my solution is wrong.

> With this change, can there be two concurrent uses of two different memory_session middlewares in one process?

No there can't be. 
I suppose that, a correct fix is to change the way how we use session middlewares. I mean like this  `Dream.memory_sessions ()`.

Reply:
Could you give some consideration to whether there is an alterative way to fix this without complicating usage in this way? And if not, then it's fine to add a `()` parameter to the session middlewares. Thank you!